File: src/main/java/Encapsulation.java
public class Encapsulation {
    public static void main(String[] args) {
        for(String card: getDeckInOrder()){
            System.out.println(card);
        }
    }

    static String[] getDeckInOrder() {
        String[] result = new String[52];
        int[][] deck = new int[52][2];

        for (int suit = 0; suit < 4; suit++) {
            for (int j = 0; j < 13; j++) {
                deck[suit*13+j] = new int[]{suit, j};
            }
        }

        int cardNumber = 0;
        for (int[] card : deck) {
            String faceValue;
            switch (card[1]){
                case 0: faceValue = "ace"; break;
                case 1:
                case 2:
                case 3:
                case 4:
                case 5:
                case 6:
                case 7:
                case 8:
                case 9: faceValue = Integer.toString(card[1]+1); break;
                case 10: faceValue = "jack"; break;
                case 11: faceValue = "queen"; break;
                case 12: faceValue = "king"; break;
                default: throw new IllegalArgumentException("Something went wrong " + card[1] + "is not a valid faceValue!");
            }

            String suitName;
            switch (card[0]){
                case 0: suitName = "clubs"; break;
                case 1: suitName = "diamonds"; break;
                case 2: suitName = "hearts"; break;
                case 3: suitName = "spades"; break;
                default: throw new IllegalArgumentException("Something went wrong " + card[0] + "is not a valid suitName!");
            }

            result[cardNumber] = faceValue + " of " + suitName;
            cardNumber++;
        }

        return result;
    }
}

File: exercises/python/cards/encapsulation.py
class Encapsulation:
    def __init__(self):
        pass

    def runner(self):
        encapsulation = Encapsulation()
        deck_in_order = encapsulation.getCards()
        for card in deck_in_order:
            print(card)

    def get_cards(self):
        result = [None] * 52
        deck = []
        suits = range(0, 4)
        face_values = range(0, 13)
        for suit in suits:
            for num in face_values:
                deck.append([suit, num])

        card_number = 0

        for card in deck:
            if card[1] == 0:
                face_value_name = "ace"
            elif card[1] in range(0, 10):
                face_value_name = str(card[1] + 1)
            elif card[1] == 10:
                face_value_name = "jack"
            elif card[1] == 11:
                face_value_name = "queen"
            elif card[1] == 12:
                face_value_name = "king"
                # default: throw new IllegalArgumentException("Something went wrong " + card[1] + "is not a valid faceValue!");}

            if card[0] == 0:
                suit_name = "clubs"
            elif card[0] == 1:
                suit_name = "diamonds"
            elif card[0] == 2:
                suit_name = "hearts"
            elif card[0] == 3:
                suit_name = "spades"
                # default: throw new IllegalArgumentException("Something went wrong " + card[0] + "is not a valid suit_name!");
            result[card_number] = face_value_name + " of " + suit_name
            card_number += 1

        return result
